rubyforgood/diaper
# [ ready for review ] Move Items counting from controller-helper to QueryObject

# Description

This PR extends the work set up in https://github.com/rubyforgood/diaper/pull/429 by moving the helper-within-the-controller to query objects.

**This PR should introduce no behavior changes.**

Please read https://github.com/rubyforgood/diaper/pull/429#issuecomment-396119620 for more context:
> The methods in question are being used to calculate totals on the fly as the view is getting rendered and iterating through the rows being displayed.
> 
> Wouldn't it be great if these calculations could be part of `@items` rather than being calculated in the controller or view? what if `def index` could say `@items_with_counts = ItemCountsQuery.new.rows`?
Then that weirdly tight controller-view-controller-view-interwoven mess could become more cleanly separated, and the controller and views would be easy to read. All that complicated calculations gets hidden away in a service (query) object.

# Checklist:

- [x] New and existing unit tests pass locally with my changes ("bundle exec rake"),
- [x] Title include "WIP" if work is in progress.

# Test instructions
On `master`:
1. Open http://localhost:3000/pdx_bank/items in 3 tabs
1. In each of the tabs, open one of the different tab views
  * Item List
  * Items and Quantity
  * Items, Quantity, and Location

On this branch:
1. In a new browser window, repeat the above
1. Verify that the numbers in the tables are exactly the same between your two browser windows
